### PR TITLE
compressor: fixes and tests; disable zlib isal (it's broken)

### PR DESCRIPTION
**Pull request by liewegas**

### CODE REVIEW

**Comment by liewegas** [comment]
@Ved-vampir the zlib ISA-L code is be broken.  I've disabled it in this PR.  Can you take a look?  Just revert the commit and run the unittest_compressor to see it fail.

**Comment by ifed01** [comment]
@liewegas She's at maternity leave at the moment, I'll take a look..

**Comment by liewegas** [comment]
Thanks!  In the meantime, this + the bluestore fix make the 
ceph_test_objectstore test pass for me without those compression errors.  
Although I promptly hit the stats error on fsck.

**Comment by liewegas** [comment]
@ifed01 ok to merge this in the meantime?

**Comment by ifed01** [comment]
Will review in a couple hours...

**Comment by ifed01** [comment]
@liewegas see PR #11351 for statfs fix

File: src/test/compressor/test_compression.cc [hunk 1]
```diff
@@ -0,0 +1,153 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
```
Comment (ifed01) [inline src/test/compressor/test_compression.cc]: Test cases you removed were Intended for plugin load verification only. There is a couple more named test_compression_snappy (_zlib) that verify compression/decompression functionality. IMHO worth combining into a single file too.

**Comment by liewegas** [comment]
yeah, i was about to do that but the zlib one has isa-related tests so i 
deferred for now.  i'll look again

File: src/test/compressor/test_compression.cc [hunk 2]
```diff
@@ -27,10 +27,30 @@
...
+    plugin = GetParam();
+    size_t pos = plugin.find('/');
+    if (pos != std::string::npos) {
+      string isal = plugin.substr(pos + 1);
+      plugin = plugin.substr(0, pos);
+      if (isal == "isal") {
+	g_conf->set_val("compressor_zlib_isal", "true");
+	g_ceph_context->_conf->apply_changes(NULL);
+      } else if (isal == "noisal") {
+	g_conf->set_val("compressor_zlib_isal", "false");
```
Comment (ifed01) [inline src/test/compressor/test_compression.cc]: Undo to default on teardown?

**Comment by liewegas** [comment]
updated

File: src/test/compressor/test_compression.cc [hunk 3]
```diff
@@ -48,6 +49,13 @@ class CompressionTest : public ::testing::Test,
       }
     }
     cout << "[plugin " << plugin << " (" << GetParam() << ")]" << std::endl;
+
+    // note for later
+    old_zlib_isal = g_conf->compressor_zlib_isal;
```
Comment (ifed01) [inline src/test/compressor/test_compression.cc]: Move upward prior to applying new values?

**Comment by ifed01** [comment]
LGTM except old_zlib_isal setup.

**Pull request merged**